### PR TITLE
combine null and empty choices in UCR / report builder filters

### PR DESCRIPTION
**Pull request by czue**
Related to https://docs.google.com/document/d/1cRFJ3AKGHnJRO2sLaTEOW4o9Z-jOYejONzv_jjW6X8Y/edit

##### SUMMARY

This is a follow up to https://github.com/dimagi/commcare-hq/pull/27645 and merges the "missing" and "empty" values into a single value called "blank" which checks for both null and empty (string) data.

Review by commit.

### CODE REVIEW

File: corehq/apps/userreports/reports/filters/choice_providers.py [hunk 1]
```diff
@@ -187,12 +190,23 @@ def get_choices_for_known_values(self, values, user):
...
         return Choice(value, value)
 
+    @staticmethod
+    def _deduplicate_and_sort_choices(choices):
+        return list(sorted(DataSourceColumnChoiceProvider._deduplicate_choices(choices),
+                           key=lambda choice: choice.display))
+
+    @staticmethod
+    def _deduplicate_choices(choices):
+        found_values = set()
```
Comment (sravfeyn) [inline corehq/apps/userreports/reports/filters/choice_providers.py:204]: `nit` Could be simplified to 
```
return {choice.value: choice for choice in choices}.values()
```

**Review by sravfeyn — APPROVED** [review]

File: corehq/apps/userreports/reports/filters/choice_providers.py (hunk 1 above)
Comment (czue) [reply corehq/apps/userreports/reports/filters/choice_providers.py]: clever - thanks. https://github.com/dimagi/commcare-hq/pull/27710/commits/16935183f820bb0d745bb5f54148984f70d656ad

**Comment by orangejenny** [comment]
nice!

**Review by esoergel — COMMENTED** [review]
Thanks for the followup - I didn't realize how involved a change this would be.

**Pull request merged**